app/v1/connectors/ListBsasConnector.scala

```diff
@@ -41,7 +41,12 @@ class ListBsasConnector @Inject()(val http: HttpClient,
       "taxYear" -> request.taxYear.toString,
       "incomeSourceIdentifier" -> request.incomeSourceIdentifier,
       "identifierValue" -> request.identifierValue
-    ).filterNot { case (_, v) => v.isInstanceOf[Option[String]] && v.asInstanceOf[Option[String]].isEmpty}
+    ).filter { // filter for ...
```

Comment:
Expanded this due to compiler warnings about type erasure.

Reply:
Just realised that I've basically re-written `.collect(...)` 🙃 I'll simplify this once David's PR gets merged in (I can get rid of the `String` checks since everything will be optional)